transport-classes-io_uring/src/main/java/io/netty/incubator/channel/uring/IOUringEventLoop.java

```diff
@@ -248,9 +261,11 @@ private void handle(int fd, int res, int flags, byte op, short data) {
             pendingWakeup = false;
             addEventFdRead(ringBuffer.ioUringSubmissionQueue());
         } else if (op == Native.IORING_OP_TIMEOUT) {
-            if (res == Native.ERRNO_ETIME_NEGATIVE) {
+            if (res == Native.ERRNO_ETIME_NEGATIVE && data == prevTimeoutGeneration) {
                 prevDeadlineNanos = NONE;
```

Comment:
we could trace loging it too (as we do we fd reuse :( ) if we assume to be an unwelcome behaviour